cognite/client/data_classes/agents/agent_tools.py

```diff
@@ -166,10 +166,12 @@ class QueryKnowledgeGraphAgentToolConfiguration(WriteableCogniteResource):
     Args:
         data_models (Sequence[DataModelInfo] | None): The data models and views to query.
         instance_spaces (InstanceSpaces | None): The instance spaces to query.
+        version (str | None): The version of the query generation strategy to use. A higher number does not necessarily mean a better query. Supported values are "v1" and "v2".
     """
 
     data_models: Sequence[DataModelInfo] | None = None
     instance_spaces: InstanceSpaces | None = None
+    version: Literal["v1", "v2"] | str | None = None
```

Comment:
Shouldn't have default values for read classes, it just makes their instantiation more error prone.
```suggestion
    data_models: Sequence[DataModelInfo] | None
    instance_spaces: InstanceSpaces | None
    version: Literal["v1", "v2"] | str | None
```